### PR TITLE
Send mail on payment failure [Z#23122835]

### PR DESCRIPTION
**Pull request by ser8phin**
Todo:

- [x] Write mail template
- [x] Call send
- [x] Add template to settings
- [x] Add logging
- [x] Write tests

### CODE REVIEW

File: src/tests/base/test_orders.py [hunk 1]
```diff
@@ -704,6 +704,47 @@ def test_sent_not_immediately_after_purchase(self):
...
+                price=Decimal("23.00"), attendee_name_parts={'full_name': "Peter"}, positionid=1
+            )
+            djmail.outbox = []
+
+    @classscope(attr='o')
+    def test_send_payment_fail_mail(self):
+        payment = self.order.payments.create(state=OrderPayment.PAYMENT_STATE_PENDING, amount=self.order.total)
+        payment.save()
+        payment.fail()
+        assert len(djmail.outbox) == 1
```
Comment (wiffbi) [inline src/tests/base/test_orders.py:737]: Would it make sense to save the len of outbox before failing and then compare whether another email was sent? That way the test should not fail, if we decide to add another email when creating a payment. So:

```suggestion
        nr_of_mails = len(djmail.outbox)
        payment.fail()
        assert len(djmail.outbox) - nr_of_mails == 1
```

File: src/tests/base/test_orders.py [hunk 2]
```diff
@@ -704,6 +704,47 @@ def test_sent_not_immediately_after_purchase(self):
...
+        assert "fail" in djmail.outbox[0].subject
+
+    @classscope(attr='o')
+    def test_no_payment_fail_mail_setting(self):
+        payment = self.order.payments.create(state=OrderPayment.PAYMENT_STATE_PENDING, amount=self.order.total)
+        payment.save()
+        payment.fail(send_mail=False)
```
Comment (wiffbi) [inline src/tests/base/test_orders.py:744]: See above regarding counting mails before fail.

**Review by wiffbi — APPROVED** [review]
Looks good, just a minor comment regarding tests and counting mails.

File: src/pretix/base/settings.py [hunk 3]
```diff
@@ -2301,6 +2301,22 @@ def unserialize(cls, s):
...
+        'default': LazyI18nString.from_gettext(gettext_noop("Payment failed for your order: {code}")),
+    },
+    'mail_text_order_payment_failed': {
+        'type': LazyI18nString,
+        'default': LazyI18nString.from_gettext(gettext_noop("""Hello,
+
+the initiated payment for your order for {event} failed.
+
+You can retry the payment and view the status of your order at
+{url}
```
Comment (raphaelm) [inline src/pretix/base/settings.py:2318]:
```suggestion
your payment attempt for your order for {event} has failed.

Your order is still valid and you can try to pay again using the same or a different payment method. Please complete your payment before {expire_date}.

You can retry the payment and view the status of your order at
{url}
```

File: src/tests/base/test_orders.py (hunk 1 above)
Comment (raphaelm) [reply src/tests/base/test_orders.py:737]: There is an ``djmail.outbox = []`` in ``setUp`` which I think should be good enough, it's very unlikely we ever send email on model creation

**Review by raphaelm — APPROVED** [review]

File: src/tests/base/test_orders.py (hunk 1 above)
Comment (ser8phin) [reply src/tests/base/test_orders.py:737]: I think that will be good enough too :)

**Pull request merged**